### ISSUE TITLE
[BUG] (usage) in bin/doom forgets to require core-cli

### ISSUE THREAD

**Issue by gagbo** [hlissner/doom-emacs#1925]
# bin/doom help errors out

`./bin/doom help` ends with a void-function error on `doom--dispatch-help`

```
./bin/doom help
Loading $HOME/.doom.d/custom.el (source)...
Loading $HOME/.doom.d/custom.el (source)...
Usage: doom [COMMAND] [ARGS...]

A command line interface for managing Doom Emacs; including
package management, diagnostics, unit tests, and byte-compilation.

This tool also makes it trivial to launch Emacs out of a different
folder or with a different private module.

Example:
  doom install
  doom help update
  doom compile :core lang/php lang/python
  doom run
  doom run -nw file.txt file2.el
  doom run -p ~/.other.doom.d -e ~/.other.emacs.d -nw file.txt

Options:
  -h --help		Same as help command
  -d --debug		Turns on doom-debug-mode (and debug-on-error)
  -e --emacsd DIR	Use the emacs config at DIR (e.g. ~/.emacs.d)
  -i --insecure		Disable TLS/SSL validation (not recommended)
  -l --local DIR	Use DIR as your local storage directory
  -p --private DIR	Use the private module at DIR (e.g. ~/.doom.d)
  -y --yes		Auto-accept all confirmation prompts

Symbol’s function definition is void: doom--dispatch-help
```

# Origin
In the cond checking if the `(usage)` function should be called, there is no `(require 'core-cli)` although `(usage)` actually calls  
https://github.com/hlissner/doom-emacs/blob/8527897c8849b0a2b91750219a75a17a3806c8d7/bin/doom#L99

# Solution
I think that either : 
- `core-cli` should be required in the first cond case,
- `usage` should require `core-cli` itself
- stop using `core-cli` in usage.

I'm more for the second solution, since I discovered the bug running `doom help` hoping to get all the available commands.

**Comment by hlissner** [comment]
This should be resolved as of 873fc5c. Let me know if that isn't the case and I'll reopen this issue. Thanks for bringing it to my attention!

**Comment by jpnadas** [comment]
I think some of the most recent commits broke this again, as I am getting the error when running `doom refresh` 

I reverted to https://github.com/hlissner/doom-emacs/commit/27bf3c85b5fe821f1ea8dde00cad9437aa733cc4 and it worked again.

**Comment by hlissner** [comment]
@jpnadas That should be fixed in 11676b0. Please update and let me know if that isn't the case.

**Comment by jpnadas** [comment]
> @jpnadas That should be fixed in [11676b0](https://github.com/hlissner/doom-emacs/commit/11676b01537bb445defe8dfe34ea9c637cae0d72). Please update and let me know if that isn't the case.

Updated and it is working again. Thanks.
